### PR TITLE
Fix clippy::use_self in extension modules

### PR DESCRIPTION
**Pull request by filnet**

### CODE REVIEW

File: ash/src/extensions/mod.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![deny(clippy::use_self)]
 pub mod experimental;
```
Comment (MarijnS95) [inline ash/src/extensions/mod.rs:2]: Doesn't `cargo fmt` add a newline between these?

File: ash/src/extensions/experimental/amd.rs [hunk 2]
```diff
@@ -40,6 +40,10 @@ vk_bitflags_wrapped!(
     Flags
 );
 impl fmt::Debug for GpaSqShaderStageFlags {
+    // ignore clippy::use_self false positives
+    // changing GpaSqShaderStageFlags::PS.0 to Self::PS.0 as suggested by clippy generates:
+    // error[E0401]: can't use generic parameters from outer function
+    #![allow(clippy::use_self)]
```
Comment (MarijnS95) [inline ash/src/extensions/experimental/amd.rs:46]: Yeah, that's a bit unfortunate. I don't think you intended to use `#!` to apply it to the surrounding `impl fmt::Debug for GpaSqShaderStageFlags` scope? Unfortunately it doesn't seem like `#[allow(clippy::use_self)]` can be applied to just the statement or the `fn fmt` function, it has to be applied to the `impl` block as a whole.

```suggestion
// ignore clippy::use_self false positives
// changing GpaSqShaderStageFlags::PS.0 to Self::PS.0 as suggested by clippy generates:
// error[E0401]: can't use generic parameters from outer function
#[allow(clippy::use_self)]
impl fmt::Debug for GpaSqShaderStageFlags {
```

**Review by MarijnS95 — APPROVED** [review]
Thanks, that's exaclty what I wanted to see in https://github.com/MaikKlein/ash/pull/421#discussion_r619842077!

File: ash/src/extensions/mod.rs (hunk 1 above)
Comment (filnet) [reply ash/src/extensions/mod.rs:2]: nope

File: ash/src/extensions/experimental/amd.rs (hunk 2 above)
Comment (filnet) [reply ash/src/extensions/experimental/amd.rs]: Done.

**Pull request merged**